### PR TITLE
fixed the case where y_train is a series while X_train is a numpy array

### LINKED ISSUE #777: Handle the case "y_train is a series while X_train is a numpy array"
Python 3.9, pandas 1.5.1.

```python
from sklearn.datasets import make_classification
from flaml import AutoML
from pandas import Series


if __name__ == "__main__":

    x_train, y_train = make_classification(1279, 20)
    y_train = Series(y_train, index=range(len(y_train), 0, -1))

    automl = AutoML()
    automl.fit(X_train=x_train, y_train=y_train, X_val=x_train, y_val=y_train, task="regression", max_iter=2)
```

flaml/automl.py:290: FutureWarning: The behavior of `series[i:j]` with an integer-dtype index is deprecated. In a future version, this will be treated as *label-based* indexing, consistent with e.g. `series[i]` lookups. To retain the old behavior, use `series.iloc[i:j]`. To get the future behavior, use `series.loc[i:j]`

Solution: make the indentation of https://github.com/microsoft/FLAML/blob/b63f1bc8b8b511932d3c519e0e214bd706e94068/flaml/automl.py#L1276-L1277
one level left.

### PR DESCRIPTION
**Pull request by evensure**
## Why are these changes needed?

To omit display of FutureWarning in cases where y_train is a Series.

## Related issue number

Closes #777 

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by ZviBaratz** [comment]
Should this be tested?

**Review by int-chaos — APPROVED** [review]

**Comment by sonichi** [comment]
> Should this be tested?

Yes. @evensure could you add a test?

**Review by qingyun-wu — APPROVED** [review]

**Review by qingyun-wu — CHANGES_REQUESTED** [review]
To add a test

**Comment by evensure** [comment]
OK, challenge accepted. I will figure this out
